### PR TITLE
Fix unused parameter warnings.

### PR DESCRIPTION
**Pull request by aloukissas**

### CODE REVIEW

**Comment by siying** [comment]
I'm fine with the code part of the change. However, by removing the parameter names, we will make it harder for users to understand what those parameters are. For example,

    -  virtual bool KeyMayExist(const ReadOptions& options,
    -                           ColumnFamilyHandle* column_family, const Slice& key,
    -                           std::string* value, bool* value_found = nullptr) {
    +  virtual bool KeyMayExist(const ReadOptions&,
    +                           ColumnFamilyHandle*, const Slice&,
    +                           std::string*, bool* value_found = nullptr) {

Users don't know the third parameter is key and the forth is value. So you may need to do something like:

     virtual bool KeyMayExist(const ReadOptions&,
                             ColumnFamilyHandle*, const Slice& /* key */,
                             std::string* /* value */, bool* value_found = nullptr) {

if you want to get the commit in.

By the way, where did you get the warnings?

**Comment by aloukissas** [comment]
@siying agreed - in my project I include rocksdb/db.h and build with -Werror -Wunused-parameter, which fails with the current code; the rocksdb makefile also has -Wno-unused-parameter (which IMO should not be the case)

File: include/rocksdb/listener.h [hunk 1]
```diff
@@ -180,7 +179,8 @@ class EventListener {
   // @param ci a reference to a CompactionJobInfo struct. 'ci' is released
   //  after this function is returned, and must be copied if it is needed
   //  outside of this function.
-  virtual void OnCompactionCompleted(DB *db, const CompactionJobInfo& ci) {}
+  virtual void OnCompactionCompleted(DB* /*db*/,
+      const CompactionJobInfo& /*ci*/) {}
```
Comment (siying) [inline include/rocksdb/listener.h]: Sometimes type names are obvious and adding a comment is more confusing. I would not add comments when it is more confusing to have the parameter. Those *Info parameters probably won't need comments. Other than that, the change looks good to me.

Comment (aloukissas) [inline include/rocksdb/listener.h]: IMO this non-uniformity is confusing. Are you implying that non-language standard types don't need comments, as they're self-explanatory?

Comment (siying) [inline include/rocksdb/listener.h]: Not a big deal. Let me merge it.

**Pull request merged**